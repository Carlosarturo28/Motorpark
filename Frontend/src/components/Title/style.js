import styled from 'styled-components'

export const TitleText = styled.h1`
font-size: 28px;
font-weight: 700;
justify-content: left;
text-align: left;
margin: 0;
color: ${props => props.color};
`