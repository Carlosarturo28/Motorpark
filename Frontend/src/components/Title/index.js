import React from 'react'
import PropTypes from 'prop-types'
import { TitleText } from './style'

const Text = ({ title, color, ...textProps }) => (
  <TitleText color={color} {...textProps}>
    {title}
  </TitleText>
)

Text.propTypes = {
  title: PropTypes.string.isRequired,
  color: PropTypes.string
}

Text.defaultProps = {
    title: 'TÃ­tulo',
    color: '#fff'
}

export default Text