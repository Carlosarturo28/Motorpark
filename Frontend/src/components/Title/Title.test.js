import React from 'react';
import { shallow } from 'enzyme';
import Title from './index';

describe('Title', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Title debug />);
  
    expect(component).toMatchSnapshot();
  });

  it('should render Title correctly with given props', () => {
  const props = { title: 'This is a Test', color: '#065' };
  const component = shallow(<Title title={props.title} color={props.color} />);
  expect(component).toMatchSnapshot();
});
});