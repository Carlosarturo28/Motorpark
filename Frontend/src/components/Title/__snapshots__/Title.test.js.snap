// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render Title correctly with given props 1`] = `
<styled.h1
  color="#065"
>
  This is a Test
</styled.h1>
`;

exports[`Title should render correctly in "debug" mode 1`] = `
<styled.h1
  color="#fff"
  debug={true}
>
  TÃ­tulo
</styled.h1>
`;
